### PR TITLE
adding support for PubkeyAcceptedKeyTypes option

### PR DESCRIPTION
**Pull request by Asteao**

### CODE REVIEW

**Comment by Asteao** [comment]
@ghoneycutt Is there an issue with this Pull request? If not, is there a reason it has not been accepted?

**Comment by Asteao** [comment]
@ghoneycutt This pull request has been open for 2 months. Is there a problem with it that needs to be fixed please let me know. I would like to stop running off of my branch in my puppet environment.

File: spec/classes/init_spec.rb [hunk 1]
```diff
@@ -2164,6 +2171,18 @@
...
   end
 
+  [true,'invalid'].each do |pubkeyacceptedkeytypes|
+    context "with sshd_pubkeyacceptedkeytypes set to invalid value #{pubkeyacceptedkeytypes}" do
+      let(:params) { { :sshd_pubkeyacceptedkeytypes => pubkeyacceptedkeytypes } }
+
+      it 'should fail' do
+        expect {
+          should contain_class('ssh')
+        }.to raise_error(Puppet::Error)
```
Comment (ghoneycutt) [inline spec/classes/init_spec.rb]: This should contain the error message

```ruby
}.to raise_error(Puppet::Error,/regex for message/)
```

**Comment by ghoneycutt** [comment]
@Asteao apologies for this. I'll get the error message fixed and get this merged. Thanks for your contribution!

**Comment by ghoneycutt** [comment]
Squashed commits in PR #218

**Comment by ghoneycutt** [comment]
Thank you very much! Released in v3.50.0

**Comment by Asteao** [comment]
Thank you!

**Pull request closed without merging**